services/121-service/src/registration/services/registrations-pagination.service.ts

```diff
@@ -163,6 +163,13 @@ export class RegistrationsPaginationService {
 
     queryBuilder = this.addPaymentFilter(queryBuilder, query);
 
+    if (query?.filter?.referenceId?.includes('$ilike') === false) {
```

Comment:
```suggestion
    if (query?.filter?.referenceId && !query?.filter?.referenceId?.includes('$ilike')) {
```

Should it rather be something like this?

Reply:
It might, what is the improvement here?

Reply:
Two different things:

```ts
    if (!query?.filter?.referenceId?.includes('$ilike')) {
```

^ This is how I would've written your check (`=== true` or `=== false` are generally avoidable)

But also, I believe that your `if` would be true even if `query?.filter?.referenceId = null` but I might be wrong.

Reply:
> if (!query?.filter?.referenceId?.includes('$ilike')) {

This does not achieve the same as it would also 'respond' to undefined/null:
![image](https://github.com/user-attachments/assets/31507bad-7aca-46e9-823c-53d78add5ecf)


> I believe that your if would be true even if query?.filter?.referenceId = null but I might be wrong

I wondered the same so I asked Chatgippiti:
![image](https://github.com/user-attachments/assets/fa90b6f3-15ae-42d6-9804-c8ee3208a577)